api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -393,3 +401,40 @@ func (m *AzureManagedMachinePool) validateKubeletConfig() error {
 	}
 	return nil
 }
+
+// validateLinuxOSConfig enforces AKS API configuration for Linux OS custom configuration
+// See: https://learn.microsoft.com/en-us/azure/aks/custom-node-configuration#linux-os-custom-configuration for detailed information.
+func (m *AzureManagedMachinePool) validateLinuxOSConfig() error {
+	if m.Spec.LinuxOSConfig == nil {
+		return nil
+	}
+
+	if m.Spec.LinuxOSConfig.SwapFileSizeMB != nil {
+		if m.Spec.KubeletConfig == nil || pointer.BoolDeref(m.Spec.KubeletConfig.FailSwapOn, true) {
+			return field.Invalid(
+				field.NewPath("Spec", "LinuxOSConfig", "SwapFileSizeMB"),
+				m.Spec.LinuxOSConfig.SwapFileSizeMB,
+				"KubeletConfig.FailSwapOn must be set to false to enable swap file on nodes")
+		}
+	}
+
+	if m.Spec.LinuxOSConfig.Sysctls != nil && m.Spec.LinuxOSConfig.Sysctls.NetIpv4IPLocalPortRange != nil {
+		portRange := m.Spec.LinuxOSConfig.Sysctls.NetIpv4IPLocalPortRange
+		if portRange.First == nil || portRange.Last == nil {
+			return field.Invalid(
+				field.NewPath("Spec", "LinuxOSConfig", "Sysctls", "NetIpv4IpLocalPortRange"),
+				m.Spec.LinuxOSConfig.Sysctls.NetIpv4IPLocalPortRange,
+				"Both NetIpv4IpLocalPortRange.First and NetIpv4IpLocalPortRange.Last must be specified")
+		}
+
+		if portRange.First != nil && portRange.Last != nil {
+			if pointer.Int32Deref(portRange.First, IPv4PortRangeDefaultFirst) > pointer.Int32Deref(portRange.Last, IPv4PortRangeDefaultLast) {
```

Comment:
Used `pointer.Int32Deref(portRange.First, IPv4PortRangeDefaultFirst)` instead of `*portRange.First` to avoid any pointer dereference errors.

Reply:
I am wondering if I should keep `pointer.Int32Deref(portRange.First, IPv4PortRangeDefaultFirst)`since `portRange.First` gets nil checked at the lines above. @nojnhuh @willie-yao

Reply:
Setting defaults via CRD defaults <=> defaulting webhook.